### DISCUSSION TITLE
Debounce not working on modelable properties in some instances

### DISCUSSION THREAD

**Discussion by timvandijck** [livewire/livewire#8644] [7. Bugs (archived)]
### Livewire version

v3.5.2

### Laravel version

v11.15.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Have a parent and child component
2. Make a property in the child component modelable
3. Use wire:model.live.debounce in the parent component to bind the property of the child component
4. Change the value of the property inside the child component

### Parent component

```php
class ParentComponent extends Component
{
    public string $childTitle = '';

    public function render()
    {
        return view('livewire.parent-component');
    }
}
```

```html
<div>
    <h1>Parent Component</h1>

    <p>
        <strong>Childs title: </strong> {{$childTitle}}
    </p>

    <livewire:child-component wire:model.live.debounce.1000ms="childTitle" />
</div>
```

### Child Component

```php
class ChildComponent extends Component
{
    #[Modelable]
    public string $title = '';

    public function render()
    {
        return view('livewire.child-component');
    }
    
    public function add()
    {
        $this->title .= fake()->word() . ' ';
    }
}
```

```html
<div>
    <h2>Child Component</h2>

    <input type="text" wire:model.live.debounce.1000ms="title">

    <p>{{$title}}</p>

    <button wire:click="add">Add</button>
</div>
```

### Repository or Wirebox URL

https://github.com/timvandijck/livewire-modelable-live-debounce

### Additional information

Debounce works perfectly when the property is updated in the child component by binding to an input. 
When you modify the property in an other way, like clicking a button, debounce is not taken into account.

I added a gif in the demo repository to demonstrate the issue (although it doesn't play at full speed inside Github it seems).

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@timvandijck debounce only works on input events, so it won't work if the property is being changed in any other way. I would suggest ensuring that your `add` button gets disabled while a network request is out (using `wire:loading`) to ensure it can't be clicked over and over.

Hope this helps!

**Reply by Kinmixoid** [reply]
Is there any chance that this will be changed at some point? Modellable components is such a powerful feature, but without debouncing it could only really work with non-live components.
